src/__tests__/App.test.tsx

```diff
@@ -0,0 +1,115 @@
+import "@testing-library/jest-dom";
+import fetchMock from "jest-fetch-mock";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+import App from "../App";
+
+// Before each test, reset and enable fetch mocking
+beforeEach(() => {
+  fetchMock.resetMocks();
+  fetchMock.enableMocks();
+});
+
+describe("App", () => {
+  it("should form not be in the document initially", () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    expect(screen.queryByRole("input")).not.toBeInTheDocument();
+  });
+
+  it("should open form when CTA button is clicked", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    const newCardInput = await screen.findByRole("textbox");
+    expect(newCardInput).toBeInTheDocument();
+  });
+
+  it("should not be possible to add a flashcard when front card value is empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+  });
+
+  it("should not be possible to add a flashcard when back card value is empty", async () => {
+    render(<App />);
+    fetchMock.mockResponse(JSON.stringify([]));
+
+    const ctaButton = screen.getByRole("button");
+    fireEvent.click(ctaButton);
+
+    let newCardInput = await screen.findByRole("textbox");
+    fireEvent.change(newCardInput, { target: { value: "" } });
+
+    const nextStepButton = await screen.findByText("Next");
+    fireEvent.click(nextStepButton);
+
+    let saveCardButton = screen.queryByText("Save");
+    expect(saveCardButton).not.toBeInTheDocument();
+
+    fireEvent.change(newCardInput, { target: { value: "This is front" } });
+    fireEvent.click(nextStepButton);
+
+    saveCardButton = await screen.findByText("Save");
+    expect(saveCardButton).toBeInTheDocument();
+
+    newCardInput = await screen.findByRole("textbox");
+
+    fireEvent.change(newCardInput, { target: { value: "" } });
+    fireEvent.click(saveCardButton);
+    expect(saveCardButton).toBeInTheDocument();
```

Comment:
**issue:** This does not test this scenario properly, as it will always be **Save** button visible in this step. You should test if that button is not clickable when clicked on it when back value is falsy

Reply:
same as above. should i render second form for back of the card instead of changing text values and pointing to different functions?

Reply:
Not exactly - just ensure that clicking on Save button does not do anything when back value is falsy